src/cli.ts

```diff
@@ -19,81 +19,92 @@ const tsxFlags = {
+			},
 		},
-		help: {
-			type: Boolean,
-			alias: 'h',
-			description: 'Show help',
-		},
-	},
-	help: false,
-	ignoreArgv: ignoreAfterArgument(),
-}, (argv) => {
-	if (argv.flags.version) {
-		process.stdout.write(`tsx v${version}\nnode `);
-	} else if (argv.flags.help) {
-		argv.showHelp({
-			description: 'Node.js runtime enhanced with esbuild for loading TypeScript & ESM',
-		});
-		console.log(`${'-'.repeat(45)}\n`);
-	}
+		help: false,
+		ignoreArgv: ignoreAfterArgument(),
+	}, (argv) => {
+		if (argv.flags.version) {
+			process.stdout.write(`tsx v${version}\nnode `);
+		} else if (argv.flags.help) {
+			argv.showHelp({
+				description: 'Node.js runtime enhanced with esbuild for loading TypeScript & ESM',
+			});
+			console.log(`${'-'.repeat(45)}\n`);
+		}
 
-	const childProcess = run(
-		removeArgvFlags(tsxFlags),
-		{
+		const childProcess = run(
+			removeArgvFlags(tsxFlags),
+			{
 			noCache: Boolean(argv.flags.noCache),
 			tsconfigPath: argv.flags.tsconfig,
-		},
-	);
+			},
+		);
 
-	const relaySignal = async (signal: NodeJS.Signals) => {
-		const message = await Promise.race([
-			/**
-			 * If child received a signal, it detected a keypress or
-			 * was sent a signal via process group.
-			 *
-			 * Ignore it and let child handle it.
-			 */
-			new Promise<NodeJS.Signals>((resolve) => {
-				function onKillSignal(data: { type: string; signal: NodeJS.Signals }) {
-					if (data && data.type === 'kill') {
-						resolve(data.signal);
-						childProcess.off('message', onKillSignal);
+		const relaySignal = async (signal: NodeJS.Signals) => {
+			const message = await Promise.race([
+				/**
+				 * If child received a signal, it detected a keypress or
+				 * was sent a signal via process group.
+				 *
+				 * Ignore it and let child handle it.
+				 */
+				new Promise<NodeJS.Signals>((resolve) => {
+					function onKillSignal(data: { type: string; signal: NodeJS.Signals }) {
+						if (data && data.type === 'kill') {
+							resolve(data.signal);
+							childProcess.off('message', onKillSignal);
+						}
 					}
-				}
 
-				childProcess.on('message', onKillSignal);
-			}),
-			new Promise((resolve) => {
-				setTimeout(resolve, 10);
-			}),
-		]);
+					childProcess.on('message', onKillSignal);
+				}),
+				new Promise((resolve) => {
+					setTimeout(resolve, 10);
+				}),
+			]);
 
-		/**
-		 * If child didn't receive a signal, it was sent to the parent
-		 * directly via kill PID. Relay it to child.
-		 */
-		if (!message) {
-			childProcess.kill(signal);
+			/**
+			 * If child didn't receive a signal, it was sent to the parent
+			 * directly via kill PID. Relay it to child.
+			 */
+			if (!message) {
+				childProcess.kill(signal);
+			}
+		};
+
+		if (process.send) {
+			let handledDep = false;
+			childProcess.on('message', (message: any) => {
+				if (message && message.type === 'dependency') handledDep = true;
+				else if (handledDep && (!message || message.type !== 'kill'))
+					process.send?.(message);
```

Comment:
```suggestion
					process.send(message);
```

You've asserted that it exists on L92